stablehlo/dialect/VhloBytecode.cpp

```diff
@@ -1197,6 +1215,70 @@ void VhloBytecodeInterface::write(TupleV1Type type,
   writer.writeTypes(type.getTypes());
 }
 
+//===----------------------------------------------------------------------===//
+// UniformQuantizedPerAxisV1Type
+//===----------------------------------------------------------------------===//
+
+UniformQuantizedPerAxisV1Type
```

Comment:
Let's make a better use of read/writeList. Here is the bytecode for upstream quantized per axis:


```
static ::mlir::Type readUniformQuantizedPerAxisType(MLIRContext* context, DialectBytecodeReader &reader) {
  uint64_t flags;
  Type storageType, expressedType;
  uint64_t quantizedDimension;
  int64_t storageTypeMin, storageTypeMax;
  SmallVector<double> scales;
  SmallVector<int64_t> zeroPoints;
  auto readScales = [&]() -> FailureOr<double> {
    double temp;
    if (succeeded(readDoubleAPFloat(reader, temp))) {
      return temp;
    }
    return failure();
  };
  auto readZeroPoints = [&]() -> FailureOr<int64_t> {
    int64_t temp;
    if (succeeded(reader.readSignedVarInt(temp))) {
      return temp;
    }
    return failure();
  };
  if (succeeded(reader.readVarInt(flags)) &&
      succeeded(reader.readType(storageType)) &&
      succeeded(reader.readType(expressedType)) &&
      succeeded(reader.readVarInt(quantizedDimension)) &&
      succeeded(reader.readSignedVarInt(storageTypeMin)) &&
      succeeded(reader.readSignedVarInt(storageTypeMax)) &&
      succeeded(reader.readList(scales, readScales)) &&
      succeeded(reader.readList(zeroPoints, readZeroPoints))) {
    return get<UniformQuantizedPerAxisType>(context, flags, storageType, expressedType, scales,
            zeroPoints, quantizedDimension, storageTypeMin, storageTypeMax);
  }
  return UniformQuantizedPerAxisType();
}

static void write(UniformQuantizedPerAxisType type, DialectBytecodeWriter &writer) {
  writer.writeVarInt(/* UniformQuantizedPerAxisType */ 5);
  writer.writeVarInt(type.getFlags());
  writer.writeType(type.getStorageType());
  writer.writeType(type.getExpressedType());
  writer.writeVarInt(type.getQuantizedDimension());
  writer.writeSignedVarInt(type.getStorageTypeMin());
  writer.writeSignedVarInt(type.getStorageTypeMax());
  writer.writeList(type.getScales(), [&](double type) {
    writer.writeAPFloatWithKnownSemantics(APFloat(type));
  });
  writer.writeList(type.getZeroPoints(), [&](int64_t type) {
    writer.writeSignedVarInt(type);
  });
}
```

Can we make use of this? As 1:1 as we can get it would be ideal.

Reply:
makes sense. updated. PTAL.